### PR TITLE
Multiple fixes

### PR DESCRIPTION
**Pull request by jdartigalongue**
- Add medium, large and xlarge sizes
- Rename authentication service to `authentication`
- Support custom registry configuration

### CODE REVIEW

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/700#" title="Author self-approved">jdartigalongue</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [jdartigalongue]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/apps/blackduck/sizes/v1/size.go [hunk 1]
```diff
@@ -3,13 +3,14 @@ package v1
...
+	"strings"
 )
 
 type size struct{}
 
 func (s size) GetSize(name string) map[string]*types.Size {
-	switch name {
-	case "small":
+	switch strings.ToUpper(name) {
+	case "SMALL":
```
Comment (rrati) [inline pkg/apps/blackduck/sizes/v1/size.go]: These strings really should be constants shared by sctl, ui, and operator.

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/700#" title="Author self-approved">jdartigalongue</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/700#pullrequestreview-264867878" title="Approved">rrati</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [jdartigalongue,rrati]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/apps/blackduck/sizes/v1/size.go (hunk 1 above)
Comment (rrati) [reply pkg/apps/blackduck/sizes/v1/size.go]: nm, this'll go away in the future anyway.

File: pkg/apps/blackduck/sizes/v1/size.go [hunk 2]
```diff
@@ -164,13 +165,502 @@ func (s size) GetSize(name string) map[string]*types.Size {
...
+					types.PostgresContainerName: {
+						MinCPU: 2,
+						MaxCPU: 2,
+						MinMem: 12288,
+						MaxMem: 12288,
+					},
+				},
+			},
+		}
+	case "XLARGE":
```
Comment (yashbhutwala) [inline pkg/apps/blackduck/sizes/v1/size.go]: This is `X-LARGE` in all other places: https://github.com/blackducksoftware/synopsys-operator/search?q=x-large&unscoped_q=x-large

File: pkg/apps/blackduck/components/rc/jobrunner/v1/component.go [hunk 3]
```diff
@@ -35,11 +35,10 @@ func (c *BdReplicationController) GetRc() (*components.ReplicationController, er
```
Comment (rrati) [inline pkg/apps/blackduck/components/rc/jobrunner/v1/component.go]: Probably should verify containerConfig isn't nil.  Technically could have key with nil value, possibly in error/edge cases, so should verify containerConfig isn't nil

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/700#" title="Author self-approved">jdartigalongue</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/700#pullrequestreview-264903335" title="Approved">rrati</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [jdartigalongue,rrati]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/700#" title="Author self-approved">jdartigalongue</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/700#pullrequestreview-264926238" title="LGTM">rrati</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [jdartigalongue,rrati]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**